libs/payments/stripe/src/lib/stripe.manager.ts

```diff
@@ -130,7 +148,12 @@ export class StripeManager {
   }
 
   async retrievePromotionCode(id: string) {
-    return this.client.promotionCodesRetrieve(id);
+    try {
+      const promotionCode = await this.client.promotionCodesRetrieve(id);
+      return promotionCode;
+    } catch (err) {
+      throw new PromotionCodeNotFoundError();
```

Comment:
Same regarding error as above. They should also include a message that indicates what promotion code was not found.